src/wallet/wallet.cpp

```diff
@@ -1170,6 +1177,71 @@ void CWallet::UpdateNullifierNoteMapWithTx(const CWalletTx& wtx)
                 mapSproutNullifiersToNotes[*item.second.nullifier] = item.first;
             }
         }
+
+        for (const mapSaplingNoteData_t::value_type& item : wtx.mapSaplingNoteData) {
+            if (item.second.nullifier) {
+                mapSaplingNullifiersToNotes[*item.second.nullifier] = item.first;
+            }
+        }
+    }
+}
+
+/**
+ * Update mapSaplingNullifiersToNotes, computing the nullifier from a cached witness if necessary.
+ */
+void CWallet::UpdateSaplingNullifierNoteMapWithTx(CWalletTx& wtx) {
+    LOCK(cs_wallet);
```

Comment:
This method is only called from `UpdateSaplingNullifierNoteMapForBlock` which has already called `LOCK(...)`.

Reply:
LOCK is recursive.